dask/tests/test_base.py

```diff
@@ -954,6 +954,17 @@ def test_visualize():
         visualize(x, filename=os.path.join(d, "mydask.png"))
         assert os.path.exists(os.path.join(d, "mydask.png"))
 
+        test_list = []
+        _mock_get_display_cls = dask.dot._get_display_cls
+
+        def mock_fx(format):
```

Comment:
It's possible I'm a bit confused here, but isn't the mocked behaviour for the function the same as the regular behaviour?
If that's true, do we really need to add the extra complexity of mocking? It might be a lot simpler to just add `visualize(x, filename=None)` as a line in `test_visualize` and a comment indicating that the test will fail if an error is raised by that line of code. 

(Note: that's the pytest default, a test will fail if an error is raised anywhere in it, but a comment is helpful for developers who are generally looking for assert statements to tell them what the test is about)

Reply:
@martindurant have I misunderstood something about the mocking here (entirely possible) or is it a bit overkill?

Reply:
It is the same behaviour, but this slightly odd construct makes sure that the function actually got called (as opposed to None being some kind of no-op). Could be removed for style's sake.